tests/upgrade/0.8.2-0.9.0/jiva/suite_test.go

```diff
@@ -0,0 +1,222 @@
+	_, err = ops.SCClient.Create(scObj)
+	Expect(err).To(BeNil(), "while creating storageclass {%s}", scObj.Name)
+
+	By("building a pvc")
+	pvcObj, err = pvc.NewBuilder().
+		WithName(pvcName).
+		WithNamespace(nsName).
+		WithStorageClass(scName).
+		WithAccessModes(accessModes).
+		WithCapacity(capacity).Build()
+	Expect(err).ShouldNot(
+		HaveOccurred(),
+		"while building pvc {%s} in namespace {%s}",
+		pvcName,
+		nsName,
+	)
+
+	By("creating above pvc")
+	_, err = ops.PVCClient.WithNamespace(nsName).Create(pvcObj)
+	Expect(err).To(
+		BeNil(),
+		"while creating pvc {%s} in namespace {%s}",
+		pvcName,
+		nsName,
+	)
+
+	By("verifying controller pod count ")
+	controllerPodCount := ops.GetPodRunningCountEventually(nsName, ctrlLabel, replicaCount)
+	Expect(controllerPodCount).To(Equal(replicaCount), "while checking controller pod count")
+
+	By("verifying replica pod count ")
+	replicaPodCount := ops.GetPodRunningCountEventually(nsName, replicaLabel, replicaCount)
+	Expect(replicaPodCount).To(Equal(replicaCount), "while checking replica pod count")
+
+	By("verifying status as bound")
+	status := ops.IsPVCBound(pvcName)
+	Expect(status).To(Equal(true), "while checking status equal to bound")
+
+})
+
+var _ = AfterSuite(func() {
+
+	By("deleting above pvc")
+	err := ops.PVCClient.Delete(pvcName, &metav1.DeleteOptions{})
+	Expect(err).To(
+		BeNil(),
+		"while deleting pvc {%s} in namespace {%s}",
+		pvcName,
+		nsName,
+	)
+
+	By("verifying controller pod count")
+	controllerPodCount := ops.GetPodRunningCountEventually(nsName, ctrlLabel, 0)
+	Expect(controllerPodCount).To(Equal(0), "while checking controller pod count")
+
+	By("verifying replica pod count")
+	replicaPodCount := ops.GetPodRunningCountEventually(nsName, replicaLabel, 0)
+	Expect(replicaPodCount).To(Equal(0), "while checking replica pod count")
+
+	By("deleting storageclass")
+	err = ops.SCClient.Delete(scName, &metav1.DeleteOptions{})
+	Expect(err).To(BeNil(), "while deleting storageclass {%s}", scObj.Name)
+
+	By("cleanup")
+	deleteFromURL(jobURL)
+	deleteFromURL(runtaskURL)
+	deleteFromURL(crURL)
+	deleteFromURL(rbacURL)
+	deleteFromURL(openebsURL)
+	By("waiting for maya-apiserver pod to come into running state")
+	podCount := ops.GetPodRunningCountEventually(
+		string(artifacts.OpenebsNamespace),
+		string(artifacts.MayaAPIServerLabelSelector),
+		0,
+	)
+	Expect(podCount).To(Equal(0))
+	podList, err := ops.PodClient.
+		WithNamespace("default").
+		List(metav1.ListOptions{})
+	Expect(err).ShouldNot(HaveOccurred())
+	for _, po := range podList.Items {
+		if po.Status.Phase == "Succeeded" {
+			err = ops.PodClient.Delete(po.Name, &metav1.DeleteOptions{})
+			Expect(err).To(BeNil(), "while deleting completed pods")
+		}
+	}
+})
+
+func applyFromURL(url string) {
+	unstructList, err := unstruct.FromURL(url)
+	Expect(err).ShouldNot(HaveOccurred())
+	// Applying unstructured objects
+	for _, us := range unstructList.Items {
+		if us.Object.GetName() == "jiva-upgrade-config" {
```

Comment:
This is tightly coupled with upgrade yamls.
If yamls are used as is then no issues.
However, the moment this logic tries to extract info from yaml then its tightly coupled.
I would still like to let tests build the `ConfigMap` & `Job` versus being dependant on manipulating yamls at runtime